### DISCUSSION TITLE
[Feature Request] Auto detect Spammers

### DISCUSSION THREAD

**Discussion by georgewoodall82** [ThioJoe/YT-Spammer-Purge#83] [Ideas]
A feature to auto detect spammers by how many comments they have posted or if the comments are the same.

For example, if ALOT of replies are made that have the same contents or are made by the same user, it will delete all comments/replies with those contents and from that user.

**Comment by ThioJoe** [comment]
Yea this is definitely something I've thought about and will add at some point

**Comment by Gowixx** [comment]
We could also create an api using `quart` or `flask` which automatically adds the channel to a list of blacklists if verified by a staff. If they get verified, it would automatically check for those banned channels and delete their comments.

**Reply by cjr60013255** [reply]
Yes there should be a blacklist library for channel users like there is a whitelist already, that way people who are already in the blacklist will automatically set for deletion/report/etc. Although I believe that an online list would be something that could happen in the future, I am sure the implementation of a local blacklist would be helpful for repeated offenders, or channels that are 100% scamming/spamming accounts.

**Comment by KendallDoesCoding** [comment]
Yeah, this is indeed a really good idea, there can be keywords like:

1. WHATSAPP ME
2. BITCOIN
3. MONEY
4. CRYPTO

And, a lot of other keywords which scammers use to auto detect scammers, this is really a good idea!

**Comment by ThioJoe** [comment]
Version 2.0 now has a "smart mode" which pretty much catches all whatsapp scammers by literally checking all unicode lookalike numbers. It also has an extremely low false positive rate. I could add more criteria to account for other types of scammer bots though.

**Reply by KendallDoesCoding** [reply]
> Version 2.0 now has a "smart mode" which pretty much catches all whatsapp scammers by literally checking all unicode lookalike numbers. It also has an extremely low false positive rate. I could add more criteria to account for other types of scammer bots though.

That's really cool!
